src/creatures/players/cast/cast_viewer.cpp

```diff
@@ -0,0 +1,1533 @@
+			sendChannelMessage("", fmt::format("New record: {} people are watching your livestream now.", std::to_string(m_castLiveRecord)), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+		}
+	}
+}
+
+void CastViewer::removeViewer(ProtocolGame_ptr client, bool spy) {
+	auto it = m_viewers.find(client);
+	if (it == m_viewers.end()) {
+		return;
+	}
+
+	auto mit = std::find(m_mutes.begin(), m_mutes.end(), it->second.first);
+	if (mit != m_mutes.end()) {
+		m_mutes.erase(mit);
+	}
+
+	if (!spy) {
+		sendChannelMessage("", fmt::format("{} has left the cast.", it->second.first), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+	}
+
+	m_viewers.erase(it);
+}
+
+void CastViewer::handle(ProtocolGame_ptr client, const std::string &text, uint16_t channelId) {
+	if (!m_owner) {
+		return;
+	}
+
+	auto sit = m_viewers.find(client);
+	if (sit == m_viewers.end()) {
+		return;
+	}
+
+	const int64_t &now = OTSYS_TIME();
+	if (client->m_castCooldownTime + 5000 < now) {
+		client->m_castCooldownTime = now, client->m_castCount = 0;
+	} else if (client->m_castCount++ >= 3) {
+		client->sendTextMessage(TextMessage(MESSAGE_STATUS, fmt::format("Please wait a {} seconds to send another message.", ((client->m_castCooldownTime + 5000 - now) / 1000) + 1)));
+		return;
+	}
+
+	bool isCastChannel = channelId == CHANNEL_CAST;
+	if (text[0] == '/') {
+		std::vector<std::string> CommandParam = explodeString(text.substr(1, text.length()), " ", 1);
+		if (strcasecmp(CommandParam[0].c_str(), "show") == 0) {
+			auto viewersNames = std::views::transform(m_viewers, [](const auto &pair) {
+				return pair.second.first;
+			});
+			std::string messageViewer = fmt::format("{} spectator{}, {}.", m_viewers.size(), m_viewers.size() > 1 ? "s" : "", fmt::join(viewersNames.begin(), viewersNames.end(), ", "));
+
+			client->sendTextMessage(TextMessage(MESSAGE_STATUS, messageViewer));
+		} else if (strcasecmp(CommandParam[0].c_str(), "name") == 0) {
+			if (CommandParam.size() > 1) {
+				if (CommandParam[1].length() > 2) {
+					if (CommandParam[1].length() < 18) {
+						bool found = false;
+						for (auto it = m_viewers.begin(); it != m_viewers.end(); ++it) {
+							if (strcasecmp(it->second.first.c_str(), CommandParam[1].c_str()) != 0) {
+								continue;
+							}
+
+							found = true;
+							break;
+						}
+
+						if (!found) {
+							if (isCastChannel) {
+								sendChannelMessage("", fmt::format("{} was renamed to {}.", sit->second.first, CommandParam[1]), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+							}
+
+							auto mit = std::find(m_mutes.begin(), m_mutes.end(), asLowerCaseString(sit->second.first));
+							if (mit != m_mutes.end()) {
+								(*mit) = asLowerCaseString(CommandParam[1]);
+							}
+
+							sit->second.first = CommandParam[1];
+						} else {
+							client->sendTextMessage(TextMessage(MESSAGE_FAILURE, "There is already someone with that name."));
+						}
+					} else {
+						client->sendTextMessage(TextMessage(MESSAGE_FAILURE, "It is not possible very long name."));
```

Comment:
```suggestion
						client->sendTextMessage(TextMessage(MESSAGE_FAILURE, "Name must be shorter than 18 letters."));
```

("18 characters" would be more correct, but I figured "letter" is easier for the target audience to understand)